### PR TITLE
Implement FileSavePicker for Gtk

### LINKED ISSUE #6126: [GTK] FileSavePicker
## What would you like to be added:

Support for `FileSavePicker` on GTK

## Why is this needed:

💾

## For which Platform:

- [ ] iOS
- [ ] Android
- [ ] WebAssembly
- [ ] WebAssembly renderers for Xamarin.Forms
- [ ] macOS
- [ ] Skia
  - [ ] WPF
  - [x] GTK (Linux)
  - [ ] Tizen
- [ ] Windows
- [ ] Build tasks
- [ ] Solution Templates

### PR DESCRIPTION
**Pull request by Youssef1313**
GitHub Issue (If applicable): #6126

## PR Type

What kind of change does this PR introduce?

- Feature

## What is the current behavior?

`FileSavePicker` is not implemented for Gtk

## What is the new behavior?

`FileSavePicker` is now implemented for Gtk

## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [x] Contains **NO** breaking changes
- [x] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [x] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

## Other information

Fixes #6126

### CODE REVIEW

**Comment by Youssef1313** [comment]
I'm not sure how to write tests or run existing tests if any for this. :(

File: src/Uno.UI.Runtime.Skia.Gtk/Extensions/Storage/Pickers/FileSavePickerExtension.cs [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+			FileChooserDialog dialog = new FileChooserDialog(
+			"Save File",
+			GtkHost.Window,
+			FileChooserAction.Save,
+			"Cancel", ResponseType.Cancel,
+			commitText, ResponseType.Accept);
+
+			dialog.SelectMultiple = false;
+			dialog.SetFilename(_picker.SuggestedFileName);
+			dialog.SetCurrentFolder(PickerHelpers.GetInitialDirectory(_picker.SuggestedStartLocation));
```
Comment (Youssef1313) [inline src/Uno.UI.Runtime.Skia.Gtk/Extensions/Storage/Pickers/FileSavePickerExtension.cs:39]: I'm not checking `FileTypeChoices` here, which most likely is a bug. Going to fix along with running/writing unit tests. Any hints for how to test? Thanks!

**Comment by MartinZikmund** [comment]
@duke7553 can you check this one please :-) ?

**Comment by duke7553** [comment]
@MartinZikmund @Youssef1313 I can't properly evaluate this within the FileSavePicker tests from the SamplesApp because of #6334, but the code looks good to me. I think all of the essential properties for FileSavePicker were covered in this PR!

**Review by duke7553 — APPROVED** [review]

**Comment by duke7553** [comment]
(Didn't mean to approve on behalf of the group)

**Review by MartinZikmund — APPROVED** [review]

**Comment by MartinZikmund** [comment]
@Youssef1313 have you also hit the #6334 while testing this?

**Comment by Youssef1313** [comment]
> @Youssef1313 have you also hit the #6334 while testing this?

I didn't actually get how the testing works in the repo tbh. So I don't know if there are existing tests somewhere that needs to be updated to include the extra platform, or if this requires new tests. I'm very new to the repo :smile:

**Comment by jeromelaban** [comment]
> > @Youssef1313 have you also hit the #6334 while testing this?
> 
> I didn't actually get how the testing works in the repo tbh. So I don't know if there are existing tests somewhere that needs to be updated to include the extra platform, or if this requires new tests. I'm very new to the repo 😄

Testing this kind of feature needs to be done manually at this point, we don't automate the skia part yet.

**Comment by Youssef1313** [comment]
> > > @Youssef1313 have you also hit the #6334 while testing this?
> > 
> > 
> > I didn't actually get how the testing works in the repo tbh. So I don't know if there are existing tests somewhere that needs to be updated to include the extra platform, or if this requires new tests. I'm very new to the repo 😄
> 
> Testing this kind of feature needs to be done manually at this point, we don't automate the skia part yet.

Is there a documentation in the repo on how to do manual testing?

**Comment by MartinZikmund** [comment]
@Youssef1313 You can launch `SamplesApp.Skia.Gtk` and then open `Windows.Storage` category in the left menu. There should be `FileSavePickerTests` page, where you can test the API along with various configuration options 👍

**Comment by Youssef1313** [comment]
@MartinZikmund The description there says:

image

But looks like it's working.

image


The only thing is that file type choices seems to be ignored. But https://github.com/unoplatform/uno/pull/6544#issuecomment-887096876 saying that all the needed properties are covered. So I'm confused.

**Comment by Youssef1313** [comment]
Actually I just hit #6334

image

but doesn't seem to be consistent.

**Comment by MartinZikmund** [comment]
> Actually I just hit #6334
> 
> image
> 
> but doesn't seem to be consistent.

Very odd 🤔🤔🤔. I really have no clue why this is happening 🤔.

**Comment by Youssef1313** [comment]
Ping @duke7553 for https://github.com/unoplatform/uno/pull/6544#issuecomment-891971540.

**Comment by Youssef1313** [comment]
I just tried, and I'm having an issue supporting `FileTypeChoices` anyway. So probably this is good to merge without `FileTypeChoices` and fix that as a followup?

cc @duke7553 @jeromelaban @MartinZikmund

**Comment by jeromelaban** [comment]
If `FileTypeChoices` is not working for now, let's open another issue and let get this merged without it.

**Comment by Youssef1313** [comment]
Re-trigerring a new build just in case :)

**Comment by Youssef1313** [comment]
@jeromelaban Is this ready to merge?

**Pull request merged**